utility/utils.py

```diff
@@ -649,8 +649,7 @@ def run_cmd(cmd, **kwargs):
         **kwargs
     )
     log.debug(f"CMD output: {r.stdout.decode()}")
-    if r.stderr:
-        log.error(f"CMD error:: {r.stderr.decode()}")
+    log.debug(f"CMD error: {r.stderr.decode()}")
```

Comment:
I think that 'if' statement can remain as no need to write to log if no stderr exist. WDYT?

Reply:
The only change I see is removing the *if* statement and I agree there is no need for any change here.

Reply:
@vasukulkarni `log.error` -> `log.debug`

Reply:
@clacroix12 @petr-balogh @vasukulkarni I think there is no need for this change for several reasons:
1) We are the only QE that will analyze those tests and this is well known that in case of an error, we should use try: except:
2) there is nothing harm in printing the error to the console - the error is real and if we are expecting for error it doesn't mean we should log it in debug level

Reply:
@petr-balogh 
> I think that 'if' statement can remain as no need to write to log if no stderr exist. WDYT?

Explicit loogging of empty stderr in debug level is more reasonable and easier to read for people who doesn't know the internals of the utils here.

Reply:
@RazTamir I **strongly disagree** with you here. The code which we are talking about is in general ` run_cmd()` function, and for this reason, we don't know if stderr output can be interpreted int as an error. This could be done in a specific context only.

The argument that since we are the only ones who reads these logs doesn't mean it's good that the choices in logging doesn't make sense. Neither it's an evidence that current design is helpful.

On the other hand, the current design is not useful in the other use case neither, when I'm writing a code which should fail if stderr is present, because it's current design doesn't allow me to do that.

This all means that currently, we will see ERROR log line reported for stderr in both cases:

 * when this is expected or doesn't indicate any error for the test
 * when this is a problem, actual error

It should be clear that this clearly is not a good design.

Reply:
This mean that I can't use this function for writing a tests anyway.

Reply:
I think that we can have another function which will return just tuple of (rc, stdout, stderr) and not raise any exception and will allow TC to process those data with specific context and react on RC, OUT and ERR as they need. @mbukatov is this what you mean?

Reply:
Hi @mbukatov ,
I'll be happy to have a BJ meeting on that PR as I don't fully understand the issues you are facing

Reply:
I quickly discussed this with @RazTamir, and we reached to following suggestion for this pull request to be merged:

* when the retcode is nonzero, report the stderr using ERROR level
* when the retcode is zero, log the stderr using DEBUG level

This will address the problem that we sometimes report ERROR log lines when it's not appropriate from the test case point of view. And on the other hand, it will keep reporting ERROR log lines when there is most likely some problem.

I understand that the current design is not ideal, and one solution could be to have another function as @petr-balogh proposes in https://github.com/red-hat-storage/ocs-ci/pull/287#discussion_r298087904. That said, I would like to find a minimal way for this pull request to be merged now, so that we improve the code a bit right immediately.

Reply:
> I quickly discussed this with @RazTamir, and we reached to following suggestion for this pull request to be merged:
> 
> * when the retcode is nonzero, report the stderr using ERROR level
> * when the retcode is zero, log the stderr using DEBUG level
WARNING level not DEBUG
so something like:
```python
if r.stderr and not r.returncode:
    log.warning(f"CMD error: {r.stderr.decode()}")
elif r.returncode :
    raise CommandFailed()
```
> 
> This will address the problem that we sometimes report ERROR log lines when it's not appropriate from the test case point of view. And on the other hand, it will keep reporting ERROR log lines when there is most likely some problem.
> 
> I understand that the current design is not ideal, and one solution could be to have another function as @petr-balogh proposes in [#287 (comment)](https://github.com/red-hat-storage/ocs-ci/pull/287#discussion_r298087904). That said, I would like to find a minimal way for this pull request to be merged now, so that we improve the code a bit right immediately.

Reply:
Well, actually, when we are talking about this and you decided to go this way: if RC is 0 and there is an stderr, that is suspicious and might deserve ERROR log lvl, while if RC is >0, having stderr is pretty normal/expected.